librz/core/cmd/cmd_analysis.c

```diff
@@ -904,347 +993,378 @@ static void core_analysis_bytes(RzCore *core, const ut8 *buf, int len, int nops,
-				if (sp) {
-					*sp = 0;
-				}
-				char *d = rz_asm_describe(core->rasm, opname);
-				if (d && *d) {
-					pj_ks(pj, "description", d);
-				}
-				free(d);
-				free(opname);
-			}
-			pj_ks(pj, "mnemonic", mnem);
-			{
-				ut8 *mask = rz_analysis_mask(core->analysis, len - idx, buf + idx, core->offset + idx);
-				char *maskstr = rz_hex_bin2strdup(mask, size);
-				pj_ks(pj, "mask", maskstr);
-				free(mask);
-				free(maskstr);
-			}
-			if (hint && hint->opcode) {
-				pj_ks(pj, "ophint", hint->opcode);
-			}
-			if (hint && hint->jump != UT64_MAX) {
-				op.jump = hint->jump;
-			}
-			if (hint && hint->fail != UT64_MAX) {
-				op.fail = hint->fail;
-			}
-			if (op.jump != UT64_MAX) {
-				pj_kn(pj, "jump", op.jump);
-			}
-			if (op.fail != UT64_MAX) {
-				pj_kn(pj, "fail", op.fail);
-			}
-			const char *jesil = (hint && hint->esil) ? hint->esil : esilstr;
-			if (jesil && *jesil) {
-				pj_ks(pj, "esil", jesil);
-			}
-			if (op.il_op) {
-				pj_k(pj, "rzil");
-				rz_il_op_effect_json(op.il_op, pj);
-			}
-			pj_kb(pj, "sign", op.sign);
-			pj_kn(pj, "prefix", op.prefix);
-			pj_ki(pj, "id", op.id);
-			if (opexstr && *opexstr) {
-				pj_k(pj, "opex");
-				pj_j(pj, opexstr);
-			}
-			pj_kn(pj, "addr", core->offset + idx);
-			{
-				char *bytes = rz_hex_bin2strdup(buf + idx, size);
-				pj_ks(pj, "bytes", bytes);
-				free(bytes);
-			}
-			if (op.val != UT64_MAX) {
-				pj_kn(pj, "val", op.val);
-			}
-			if (op.disp && op.disp != UT64_MAX) {
-				pj_kn(pj, "disp", op.disp);
-			}
-			if (op.ptr != UT64_MAX) {
-				pj_kn(pj, "ptr", op.ptr);
-			}
-			pj_ki(pj, "size", size);
-			pj_ks(pj, "type", rz_analysis_optype_to_string(op.type));
-			{
-				const char *datatype = rz_analysis_datatype_to_string(op.datatype);
-				if (datatype) {
-					pj_ks(pj, "datatype", datatype);
-				}
-			}
-			if (esilstr) {
-				int ec = esil_cost(core, addr, esilstr);
-				pj_ki(pj, "esilcost", ec);
-			}
-			if (op.reg) {
-				pj_ks(pj, "reg", op.reg);
-			}
-			if (op.ireg) {
-				pj_ks(pj, "ireg", op.ireg);
-			}
-			pj_ki(pj, "scale", op.scale);
-			if (op.refptr != -1) {
-				pj_ki(pj, "refptr", op.refptr);
-			}
-			pj_ki(pj, "cycles", op.cycles);
-			pj_ki(pj, "failcycles", op.failcycles);
-			pj_ki(pj, "delay", op.delay);
-			const char *p1 = rz_analysis_stackop_tostring(op.stackop);
-			if (strcmp(p1, "null")) {
-				pj_ks(pj, "stack", p1);
+
+		char strsub[128] = { 0 };
+		// pc+33
+		rz_parse_subvar(core->parser, NULL,
+			core->offset + idx,
+			asmop.size, rz_asm_op_get_asm(&asmop),
+			strsub, sizeof(strsub));
+		ut64 killme = UT64_MAX;
+		if (rz_io_read_i(core->io, op.ptr, &killme, op.refptr, be)) {
+			core->parser->subrel_addr = killme;
+		}
+		// 0x33->sym.xx
+		char *p = strdup(strsub);
+		if (p) {
+			rz_parse_filter(core->parser, addr, core->flags, hint, p,
+				strsub, sizeof(strsub), be);
+			free(p);
+		}
+		pj_o(pj);
+		pj_ks(pj, "opcode", rz_asm_op_get_asm(&asmop));
+		if (!*strsub) {
```

Comment:
RZ_STR_ISEMPTY

Reply:
Thanks for your reviewing. This leads to `warning: the address of ‘strsub’ will always evaluate as ‘true’ [-Waddress]   64 | #define RZ_STR_ISEMPTY(x)    (!(x) || !*(x))` in compile. Is that ok?

Reply:
keep it in this way then.